### PR TITLE
feat(store): support Angular 8, TypeScript 3.4

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[x] Other... Please describe:
```

## What is the current behavior?

Only support Angular 7, TypeScript 3.1

## What is the new behavior?

Support Angular 8, TypeScript 3.4

After upgrading to Angular 8, TypeScript 3.4 all types were broken and tests did not pass!
In order to avoid breaking change, the types were set to `any`.
In the next pull request, we need to apply a new typing to support state operators.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by fetters5** [comment]
I see you've referenced jest 8.0.1, you may want to wait for PR #[330](https://github.com/meltedspark/angular-builders/pull/330) to be merged or you may not be able to run tests

**Comment by whichwit** [comment]
That referenced PR actually bumped jest builder to 8.0.2, not sure how much it matters for this PR.

**Comment by splincode** [comment]
@arturovt What do you think about state operators and their typing?

**Comment by markwhitfeld** [comment]
Not too happy about the `any` typing of the state operators. 
Could you explain the errors that were causing this to fail?

**Comment by arturovt** [comment]
@splincode 
I'm not a fan of it. As I know you've been fighting for the good typings support of this library. I wouldn't like state operators to support `any` type.

**Comment by splincode** [comment]
@arturovt Um, by no means, I didn’t want to worsen the types, I just tried to at least pass the tests and that the assembly worked. As you understand these operators you have developed, and typing does not work for your operators in TS 3.4, a compilation error occurs.

File: packages/store/operators/src/compose.ts [hunk 1]
```diff
@@ -1,7 +1,16 @@
 import { StateOperator } from '@ngxs/store';
 
-export function compose<T>(...operators: StateOperator<T>[]) {
```
Comment (arturovt) [inline packages/store/operators/src/compose.ts]: image

File: packages/store/operators/src/append.ts [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+ * In the next pull request, we need to apply a new typing to support state operators.
+ * TODO: Need to fix types
+ */
+
 /**
  * @param items - Specific items to append to the end of an array
  */
-export function append<T>(items: T[]) {
-  return function appendOperator(existing: Readonly<T[]>): T[] {
+export function append<T>(items: T[]): StateOperator<any> {
```
Comment (arturovt) [inline packages/store/operators/src/append.ts]: image

**Comment by arturovt** [comment]
If tests don't pass - this doesn't mean that we have to change everything to `any` type, just because "to make tests happy". This means we're doing something wrong.

File: packages/router-plugin/package.json [hunk 3]
```diff
@@ -6,8 +6,8 @@
   "sideEffects": true,
   "peerDependencies": {
     "@ngxs/store": "^0.0.0",
-    "@angular/core": ">=5.0.0 <8.0.0",
-    "@angular/router": ">=5.0.0 <8.0.0",
+    "@angular/core": ">=5.0.0 <9.0.0",
+    "@angular/router": ">=5.0.0 <9.0.0",
```
Comment (kuncevic) [inline packages/router-plugin/package.json]: As per this twit https://twitter.com/stephenfluin/status/1129429898047643648 suggested to set peer dependency to ^8.0.0 || ^9.0.0-0.

File: packages/store/operators/src/compose.ts (hunk 1 above)
Comment (splincode) [reply packages/store/operators/src/compose.ts]: Yeah, this build is correct for me also, but not working for test

File: packages/store/operators/src/append.ts (hunk 2 above)
Comment (splincode) [reply packages/store/operators/src/append.ts]: Yeah

File: packages/router-plugin/package.json (hunk 3 above)
Comment (splincode) [reply packages/router-plugin/package.json]: Good catch

**Comment by arturovt** [comment]
@splincode What do you think if support of Angular 8 would be a part of NGXS v4?

**Comment by splincode** [comment]
@arturovt My company switched to Angular 8 and our team  switched also projects, but if NGXS does not have Angular 8 support (our company has set up a common CI, our build fall if warnings are shown when installing packages), then our architect will decide to switch to NGRX 8  to integrate with Angular 8. And then I will have to switch to NGRX in my projects due to poor support for NGXS.  I still hope that the NGXS will not keep up with current trends.  NGXS for a long time did not release anything, no new version and we are very worried about this topic.

**Comment by arturovt** [comment]
I don't mind switching to Angular 8 but without breaking typings and setting them to `any`.

**Comment by splincode** [comment]
I hope in the future NGXS will support Angular 8 and Typescript 3.4. I did everything I could.

**Comment by markwhitfeld** [comment]
@arturovt and @splincode I am currently working on the typings to hopefully resolve this.

**Pull request closed without merging**